src/Components/selectedArticlePage.jsx

```diff
@@ -1,9 +1,88 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchComments } from "../api";
+import Comments from "./comments";
+import VoteButtons from "./VoteButtons";
 
-const SelectedArticle = () => {
-    return (
-        <h1> hello </h1>
-    )
-}
+const SelectedArticle = ({ fetchArticles }) => {
+  const { articleId } = useParams();
+  const [selectedArticle, setSelectedArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [comments, setComments] = useState([]);
+  const [error, setError] = useState(null);
 
-export default SelectedArticle
\ No newline at end of file
+  const formattedDate = new Date(selectedArticle.created_at).toLocaleDateString(
+    "en-uk",
+    {
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+    }
+  );
+
+  const handleVote = (updatedArticle) => {
+    setSelectedArticle(updatedArticle);
+  };
+
+  useEffect(() => {
+    const fetchSelectedArticle = () => {
+
+      fetchArticles(articleId)
+        .then((data) => {
+          setSelectedArticle(data);
+        })
+        .catch((error) => {
+          console.error("Error fetching selected article:", error);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    };
+    const fetchArticleComments = () => {
+      fetchComments(articleId)
+        .then((data) => {
+          setComments(data);
+        })
+        .catch((error) => {
+          console.error("Error fetching comments:", error);
+        });
```

Comment:
Could you implement a loading state for your comments as well
like you've expertly done for for article